src/helper-modules/govuk-notify/govuk-notify.service.test.ts

```diff
@@ -0,0 +1,133 @@
+      expect(NotifyClient).toHaveBeenCalledWith(govUkNotifyKey);
+    });
+
+    it('calls GOV.UK Notify client sendEmail function with the correct arguments', async () => {
+      await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation, reference });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      expect(sendEmailMethodMock).toHaveBeenCalledWith(templateId, sendToEmailAddress, { personalisation, reference });
+    });
+
+    it('returns a 201 response from GOV.UK Notify when sending the email is successful', async () => {
+      const response = await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
+
+    it.each([
+      {
+        exceptionClass: BadRequestException,
+        exceptionName: 'Bad Request Exception',
+        error: 'Bad Request',
+        status: 400,
+      },
+      {
+        exceptionClass: UnauthorizedException,
+        exceptionName: 'Unauthorized Exception',
+        error: 'Unauthorized',
+        status: 401,
+      },
+      {
+        exceptionClass: ForbiddenException,
+        exceptionName: 'Forbidden Exception',
+        error: 'Forbidden',
+        status: 403,
+      },
+      {
+        exceptionClass: InternalServerErrorException,
+        exceptionName: 'Internal Server Error Exception',
+        error: 'Internal Server Error',
+        status: 500,
+      },
+    ])('throws exception $exceptionName for unexpected $status', async ({ exceptionClass, exceptionName, error, status }) => {
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(status, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(exceptionClass);
+      await expect(resultPromise).rejects.toThrow(exceptionName);
+      await expect(resultPromise).rejects.toHaveProperty('status', status);
+      await expect(resultPromise).rejects.toHaveProperty('response', { message: [errorMessage], error, statusCode: status });
+    });
+
+    it('throws generic Error exception for unexpected status', async () => {
+      const unexpectedStatus = valueGenerator.integer({ min: 900, max: 999 });
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(unexpectedStatus, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(Error);
+      await expect(resultPromise).rejects.toThrow(errorMessage);
+      await expect(resultPromise).rejects.toHaveProperty('message', errorMessage);
```

Comment:
```suggestion
      const unexpectedStatus = valueGenerator.integer({ min: 900, max: 999 });
      const notifyError = generateNotifyError(unexpectedStatus, errorMessage)
      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(notifyError));

      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });

      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
      await expect(resultPromise).rejects.toBeInstanceOf(Error);
      await expect(resultPromise).rejects.toThrow(errorMessage);
      await expect(resultPromise).rejects.toHaveProperty('message', errorMessage);
      expect(loggerError).toHaveBeenCalledTimes(1);
      expect(loggerError).toHaveBeenCalledWith(notifyError);
```

Reply:
This code worked for me for checking that the logger was called with the correct error. This only worked when the `expect(loggerError)...` statements were after the `await` statements (I still don't fully understand why this is, so I'm going to read up on it).

Reply:
Thanks, nice simple test